mirroring/mirroring.go

```diff
@@ -136,46 +137,88 @@ func VerifySignedTreeHead() error {
 	return nil
 }
 
-func ComputeNewRootHash(oldRootHash string, logProof *models.ConsistencyProof) (string, error) {
-	newRootHash, err := hex.DecodeString(oldRootHash)
+func VerifyLogConsistency(oldSize int64, oldRootHash string) error {
+	rekorClient, err := client.GetRekorClient(rekorServerURL)
 	if err != nil {
-		return "", err
+		return err
+	}
+
+	logInfo, err := GetLogInfo(rekorClient)
+	if err != nil {
+		return err
+	}
+
+	logProof, err := GetLogProof(rekorClient, &oldSize, logInfo.TreeSize)
+	if err != nil {
+		return err
+	}
+
+	oldRoot, err := hex.DecodeString(oldRootHash)
+	if err != nil {
+		return err
 	}
 
-	for _, hashStr := range logProof.Hashes {
-		hash, err := hex.DecodeString(hashStr)
+	newRoot, err := hex.DecodeString(*logInfo.RootHash)
+	if err != nil {
+		return err
+	}
+
+	proofs := make([][]byte, len(logProof.Hashes))
+	for i, h := range logProof.Hashes {
+		hash, err := hex.DecodeString(h)
 		if err != nil {
-			return "", err
+			return err
 		}
-		newRootHash = rfc6962.DefaultHasher.HashChildren(newRootHash, hash)
+		proofs[i] = hash
+	}
+
+	verifier := logverifier.New(rfc6962.DefaultHasher)
```

Comment:
I wonder if the hasher here can be loaded from a location in the rekor codebase later...

Reply:
For a short answer, I think we can do that! But to understand better, will that be something like switching to different hasher or initializing with a custom/user-defined hasher?

Reply:
Can you make a ticket about this too? maybe a question tag for it :)